src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1758,6 +1758,28 @@ private void BindArgumentsAndNames(BracketedArgumentListSyntax argumentListOpt,
             }
         }
 
+        private bool RefMustBeObeyed(bool isDelegateCreation, ArgumentSyntax argumentSyntax)
+        {
+            if (Compilation.FeatureStrictEnabled || !isDelegateCreation)
+            {
+                return true;
+            }
+
+            switch (argumentSyntax.Expression.Kind())
+            {
+                case SyntaxKind.ParenthesizedLambdaExpression:
+                case SyntaxKind.SimpleLambdaExpression:
+                case SyntaxKind.AnonymousMethodExpression:
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.ParenthesizedExpression:
```

Comment:
The legacy compiler will reject anything in a `ParenthesizedExpression` just like it does in strict mode. I made a comment in the test case.